crates/iota/src/iota_commands.rs

```diff
@@ -1245,9 +1244,9 @@ fn prompt_if_no_config(
         }
         .join(IOTA_KEYSTORE_FILENAME);
         let keystore = Keystore::from(FileBasedKeystore::new(&keystore_path)?);
-        let mut config = IotaClientConfig::new(keystore);
+        let mut config = IotaClientConfig::new(keystore).with_default_envs();
         if prompt_for_env {
-            config.add_env(prompt_for_environment(wallet_conf_path, accept_defaults)?);
+            config.set_env(prompt_for_environment(wallet_conf_path, accept_defaults)?);
```

Comment:
`cargo run client --yes new-env --alias testnet --rpc https://api.devnet.iota.cafe --graphql https://graphql.devnet.iota.cafe` (used devnet URLs on purpose here to see that it would be actually used)
this fails with `Environment config with name [testnet] already exists.`

Reply:
so, do we want to override the exiting env then?

Reply:
there you go: https://github.com/iotaledger/iota/pull/7950/commits/08444749f3dfe5aaa49af9dde02f994a1c1e4e34

but not sure there are other edge cases not covered yet.